storage/src/adb/any.rs

```diff
@@ -386,8 +386,8 @@ impl<E: RStorage + Clock + Metrics, K: Array, V: Array, H: CHasher> Any<E, K, V,
             .iter()
             .map(|op| Any::<E, _, _, _>::op_digest(hasher, op))
             .collect::<Vec<_>>();
-
-        proof.verify_range_inclusion(hasher, &digests, start_pos, end_pos, root_hash)
+        let mut elements_iter = digests.iter().map(|d| d.as_ref());
+        proof.verify_range_inclusion(hasher, &mut elements_iter, start_pos, end_pos, root_hash)
```

Comment:
Why does this need to be a mutable reference? It seems verify should be immutable?

Reply:
It calls next() on the iterator which requires it to be mutable.

Reply:
Changed this to accept an IntoIterator so the muts are no longer necessary.

Reply:
Made the iterator items AsRef<T> which, combined with the IntoIterator change, allowed me to revert most of the changes in this PR!